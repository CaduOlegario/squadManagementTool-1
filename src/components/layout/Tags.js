import React from 'react';
import {createMuiTheme, TextField, Typography} from "@material-ui/core";
import {makeStyles} from "@material-ui/core/styles";
import Autocomplete from '@material-ui/lab/Autocomplete';
import Chip from "@material-ui/core/Chip";
import Box from "@material-ui/core/Box";
import ClearIcon from '@material-ui/icons/Clear';
import {ThemeProvider} from "@material-ui/styles";
import useTheme from "@material-ui/core/styles/useTheme";

const useStyles = makeStyles(theme => ({
    root: {

    },
}));

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        palette: {
            ...theme.palette,
            secondary: {
                main: 'rgb(0, 0, 0)'
            }
        },
        overrides: {
            MuiInputBase: {
                root: {
                    fontSize: "0.8rem"
                },
            },
            MuiOutlinedInput: {
                input: {
                    padding: "10px 14px"
                }
            },
            MuiAutocomplete: {
                inputRoot: {
                    height: "114px",
                    alignContent: "baseline"
                }
            }
        }
    });

const Tags = (props) => {
    const classes = useStyles();
    const theme = useTheme();

    return (
        <ThemeProvider theme={getMuiTheme(theme)}>
            <Box mb={0.5}>
                <Typography variant="subtitle1" >
                    {props.label}
                </Typography>
            </Box>
            <Autocomplete
                {...props}
                multiple
                className={classes.root}
                options={[]}
                freeSolo
                multiline
                rows={4}
                closeIcon={<React.Fragment />}
                renderTags={(value, getTagProps) =>
                    value.map((option, index) => (
                        <Chip
                            label={option}
                            color="primary"
                            size="small"
                            clickable
                            deleteIcon={<ClearIcon />}
                            {...getTagProps({ index })} />
                    ))
                }
                renderInput={(params) => (
                    <TextField
                        {...params}
                        color="secondary"
                        variant="outlined"
                        placeholder={props.value.length === 0 && props.placeholder}
                    />
                )}
            />
        </ThemeProvider>
    )
};

export default Tags;