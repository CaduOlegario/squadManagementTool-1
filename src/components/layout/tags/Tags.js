import React from 'react';
import {createMuiTheme, TextField, Typography} from "@material-ui/core";
import Autocomplete from '@material-ui/lab/Autocomplete';
import Chip from "@material-ui/core/Chip";
import Box from "@material-ui/core/Box";
import ClearIcon from '@material-ui/icons/Clear';
import {ThemeProvider} from "@material-ui/styles";
import useTheme from "@material-ui/core/styles/useTheme";
import PropTypes from "prop-types";
import Input from "../input/Input";

const getMuiTheme = (theme) =>
    createMuiTheme({
        ...theme,
        palette: {
            ...theme.palette,
            secondary: {
                main: 'rgb(0, 0, 0)'
            }
        },
        overrides: {
            MuiInputBase: {
                root: {
                    fontSize: "0.8rem"
                },
            },
            MuiOutlinedInput: {
                input: {
                    padding: "10px 14px"
                }
            },
            MuiAutocomplete: {
                inputRoot: {
                    height: "114px",
                    alignContent: "baseline"
                }
            }
        }
    });

/** Input component that create tags. Accepts the same props from [Material-ui autocomplete free solo multiple values](https://material-ui.com/components/autocomplete/#multiple-values) */
const Tags = (props) => {
    const theme = useTheme();
    return (
        <ThemeProvider theme={getMuiTheme(theme)}>
            <Box mb={0.5}>
                <Typography variant="subtitle1">
                    {props.label}
                </Typography>
            </Box>
            <Autocomplete
                {...props}
                multiple
                options={[]}
                freeSolo
                rows={4}
                closeIcon={<React.Fragment/>}
                renderTags={(value, getTagProps) =>
                    value.map((option, index) => (
                        <Chip
                            label={option}
                            color="primary"
                            size="small"
                            clickable
                            deleteIcon={<ClearIcon/>}
                            {...getTagProps({index})} />
                    ))
                }
                renderInput={(params) => (
                    <TextField
                        {...params}
                        color="secondary"
                        variant="outlined"
                        placeholder={props.placeholder}
                    />
                )}
            />
        </ThemeProvider>
    )
};


Input.propTypes = {
    /** Input's label */
    label: PropTypes.string.isRequired,
    /** Input's placeholder */
    placeholder: PropTypes.string,
};

export default Tags;