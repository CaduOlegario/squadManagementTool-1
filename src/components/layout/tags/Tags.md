Tags example, use enter to create new tags:

```js
    <Tags
        label="Tags"
        placeholder="Fill with you tags"
    />
```