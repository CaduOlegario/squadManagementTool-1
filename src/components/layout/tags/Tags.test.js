import React from 'react';
import renderer from 'react-test-renderer';
import Tags from "./Tags";

it('renders correctly normal input', () => {
    const tree = renderer
        .create(<Tags
            label="Tags"
            placeholder="Fill with you tags"/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
