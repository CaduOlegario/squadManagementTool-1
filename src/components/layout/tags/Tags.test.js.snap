// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal input 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
    >
      Tags
    </h6>
  </div>,
  <div
    aria-expanded={false}
    aria-owns={null}
    className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon"
    label="Tags"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    placeholder="Fill with you tags"
    role="combobox"
    rows={4}
  >
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
        onClick={[Function]}
      >
        <input
          aria-activedescendant={null}
          aria-autocomplete="list"
          aria-controls={null}
          aria-invalid={false}
          autoCapitalize="none"
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          placeholder="Fill with you tags"
          required={false}
          spellCheck="false"
          type="text"
          value=""
        />
        <div
          className="MuiAutocomplete-endAdornment"
        >
          <button
            aria-label="Clear"
            className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            title="Clear"
            type="button"
          >
            <span
              className="MuiIconButton-label"
            />
          </button>
        </div>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-3"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
]
`;
